js/apps/admin-ui/src/components/dynamic/UrlComponent.tsx

```diff
@@ -0,0 +1,44 @@
+import { FormGroup } from "@patternfly/react-core";
+import { useTranslation } from "react-i18next";
+
+import { HelpItem } from "ui-shared";
+import { Controller, useFormContext } from "react-hook-form";
+import type { ComponentProps } from "./components";
+import { FormattedLink } from "../external-link/FormattedLink";
+import { convertToName } from "./DynamicComponents";
+
+import "./url-component.css";
+
+export const UrlComponent = ({
+  name,
+  label,
+  helpText,
+  defaultValue
+}: ComponentProps) => {
+  const { t } = useTranslation();
+  const { control } = useFormContext();
+  
+  return (
+    
+    <FormGroup
+      label={t(label!)}
+      fieldId={name!}
+      labelIcon={<HelpItem helpText={t(helpText!)} fieldLabelId={`${label}`} />}
+      isRequired={false}
+      className="keycloak__identity-providers__url_component"
+    >
+      <Controller
+        name={convertToName(name!)}
+        defaultValue={defaultValue}
+        control={control}
+        render={({ field }) => (
```

Comment:
You don't need this controller, because this value can't be changed

Reply:
@edewit please excuse my ignorance in React (not my area, at all) but how can I reach the `field.value` after without wrapping in `Controller`?

Reply:
Applied! (pardon me if something is still not 100% clear, feel like my 1st day on the job when using React :)